src/server/dfly_main.cc

```diff
@@ -177,37 +181,50 @@ struct VersionMonitor {
   }
 
  private:
-  void RunTask();
+  void RunTask(SSL_CTX* ssl_ctx);
 };
 
-void VersionMonitor::RunTask() {
+void VersionMonitor::Run(ProactorPool* proactor_pool) {
+  // Don't run in dev environment - i.e. where we don't build with
+  // real version number
+  if (!GetFlag(FLAGS_version_check) || kGitTag[0] != 'v')
+    return;
+
+  SSL_CTX* ssl_ctx = TlsClient::CreateSslContext();
+  if (!ssl_ctx) {
+    VLOG(1) << "Remote version - failed to create SSL context - cannot run version monitoring";
+    return;
+  }
+
+  version_fiber_ =
+      proactor_pool->GetNextProactor()->LaunchFiber([ssl_ctx, this] { RunTask(ssl_ctx); });
+}
+
+void VersionMonitor::RunTask(SSL_CTX* ssl_ctx) {
   const auto loop_sleep_time = std::chrono::hours(24);  // every 24 hours
 
   const std::string host_name = "version.dragonflydb.io";
   const std::string_view port = "443";
   const std::string resource = "/v1";
-  const std::string version_header = std::string("DragonflyDB/") + kGitTag;
+  string_view current_version(kGitTag);
 
-  SSL_CTX* context = TlsClient::CreateSslContext();
-  if (!context) {
-    VLOG(1) << "Remote version - failed to create SSL context - cannot run version monitoring";
-    return;
-  }
+  current_version.remove_prefix(1);
```

Comment:
what is this prefix that we're removing here? I though that git version is taken from the tag